src/main/java/de/dennisguse/opentracks/util/PreferencesUtils.java

```diff
@@ -318,4 +345,76 @@ public static void setDefaultExportDirectoryUri(SharedPreferences sharedPreferen
     public static boolean isDefaultExportDirectoryUri(SharedPreferences sharedPreferences, Context context) {
         return getDefaultExportDirectoryUri(sharedPreferences, context) != null;
     }
+
+    public static int getLayoutColumns(SharedPreferences sharedPreferences, Context context) {
+        return getInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, context.getResources().getInteger(R.integer.stats_custom_layout_columns_default));
+    }
+
+    public static void setLayoutColumns(SharedPreferences sharedPreferences, Context context, int columns) {
+        setInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, columns);
+    }
+
+    private static List<TypedArray> getMultiTypedArray(Context context, String key) {
+        List<TypedArray> typedArrays = new ArrayList<>();
+
+        try {
+            Class<R.array> resource = R.array.class;
+            Field field;
+            int counter = 0;
+
+            do {
+                field = resource.getField(key + "_" + counter);
+                typedArrays.add(context.getResources().obtainTypedArray(field.getInt(null)));
+                counter++;
+            } while (field != null);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return typedArrays;
+    }
+
+    @SuppressLint("ResourceType")
+    private static String buildDefaultLayout(Context context) {
+        List<TypedArray> typedArrays = getMultiTypedArray(context, "stats_custom_layout_fields_default_value");
+        return context.getString(R.string.default_activity_default) + ";"
+                + typedArrays.stream().map(i -> i.getString(0) + "," + i.getString(1)).collect(Collectors.joining(";"))
+                + ";";
+    }
+
+    public static Layout getCustomLayout(SharedPreferences sharedPreferences, Context context) {
+        String csvCustomLayout = getString(sharedPreferences, context, R.string.stats_custom_layout_fields_key, buildDefaultLayout(context));
+        List<String> csvParts = Arrays.asList(csvCustomLayout.split(";"));
+        Layout layout = new Layout(csvParts.get(0));
+        for (int i = 1; i < csvParts.size(); i++) {
+            List<String> fieldParts = Arrays.asList(csvParts.get(i).split(","));
+            layout.addField(fieldParts.get(0), fieldParts.get(1).equals("1"), fieldParts.get(2).equals("1"), fieldParts.get(0).equals(context.getString(R.string.stats_coordinates)));
+        }
+
+        return layout;
+    }
+
+    public static void setCustomLayout(SharedPreferences sharedPreferences, Context context, Layout layout) {
+        List<Layout.Field> fields = layout.getFields();
+        if (fields.size() == 0) {
+            return;
+        }
+
+        String csv = layout.getProfile() + ";" + fields.stream().map(field -> field.getKey() + "," + (field.isVisible() ? "1" : "0") + "," + (field.isPrimary() ? "1" : "0")).collect(Collectors.joining(";")) + ";";
```

Comment:
something like: `profile;fieldName,visible,primary;` might be more readable/ maintainable

may we use `public static String CSV_SEPARATOR = ";"`?

Reply:
and can use create a method in Field to parse/ write it's data?